engine/gui/src/gui.cpp

```diff
@@ -3880,6 +3879,90 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
+            Vector4 adjusted_dims = mulPerElem(parent_dims, adjust_scale);
+            Vector4 ref_size = Vector4((float) scene->m_Context->m_PhysicalWidth, (float) scene->m_Context->m_PhysicalHeight, 0.0f, 1.0f);
+            offset = (ref_size - adjusted_dims) * 0.5f;
+        }
+
+        // We calculate a new position that will be the relative position once
+        // the node has been childed to the new parent.
+        Vector3 position = screen_position - parent_m.getCol3().getXYZ();
+
+        // We need to perform the inverse of what AdjustPosScale will do to counteract when
+        // it will be applied during next call to CalculateNodeTransform.
+        // See AdjustPosScale for comparison on the steps being performed/inversed.
+        if (node->m_Node.m_XAnchor == XANCHOR_LEFT || node->m_Node.m_XAnchor == XANCHOR_RIGHT) {
+            offset.setX(0.0f);
+        }
+        if (node->m_Node.m_YAnchor == YANCHOR_TOP || node->m_Node.m_YAnchor == YANCHOR_BOTTOM) {
+            offset.setY(0.0f);
+        }
+
+        Vector3 scaled_position = position - offset.getXYZ();
+        position = mulPerElem(recipPerElem(adjust_scale.getXYZ()), scaled_position);
+
+        if (node->m_Node.m_XAnchor == dmGui::XANCHOR_LEFT || node->m_Node.m_XAnchor == dmGui::XANCHOR_RIGHT) {
+            position.setX(scaled_position.getX() / reference_scale.getX());
+        }
+
+        if (node->m_Node.m_YAnchor == dmGui::YANCHOR_BOTTOM || node->m_Node.m_YAnchor == dmGui::YANCHOR_TOP) {
+            position.setY(scaled_position.getY() / reference_scale.getY());
+        }
+        return position;
+    }
+
+    Point3 ScreenToLocalPosition(HScene scene, HNode node, const Point3& screen_position)
+    {
+        InternalNode* n = GetNode(scene, node);
+        InternalNode* parent_node = 0x0;
+        if (n->m_ParentIndex != INVALID_INDEX)
+        {
+            parent_node = &scene->m_Nodes[n->m_ParentIndex];
+        }
+        Vector3 local_position = ScreenToLocalPosition(scene, n, parent_node, Vector3(screen_position));
+        return Point3(local_position);
+    }
+
+    void SetScreenPosition(HScene scene, InternalNode* node, InternalNode* parent_node, dmVMath::Vector3 screen_position)
+    {
+        Vector3 local_position = ScreenToLocalPosition(scene, node, parent_node, screen_position);
+        node->m_Node.m_Properties[dmGui::PROPERTY_POSITION] = Vector4(local_position, 1.0f);
+        node->m_Node.m_DirtyLocal = 1;
+    }
+
+    void SetScreenPosition(HScene scene, HNode node, const Point3& screen_position)
+    {
+        InternalNode* n = GetNode(scene, node);
+        InternalNode* parent_node = 0x0;
+        if (n->m_ParentIndex != INVALID_INDEX)
+        {
+            parent_node = &scene->m_Nodes[n->m_ParentIndex];
+        }
+        SetScreenPosition(scene, n, parent_node, Vector3(screen_position));
+    }
+
     Result SetNodeParent(HScene scene, HNode node, HNode parent, bool keep_scene_transform)
```

Comment:
I got functionality from this function into two: `ScreenToLocalPosition` and `SetScreenPosition` (2 versions each, one external usage in `gui.h` and another one for internal usage)

Reply:
Pls make the internal functions "static"